### PR TITLE
builder: split out builder into a dedicated crate

### PR DESCRIPTION
**Pull request by jiangliu**
Split out builder into a dedicate nydus-builder crate, to reduce dependencies of the nydus-rafs crate.

## Relevant Issue (if applicable)
_If there are Issues related to this PullRequest, please list it._

## Details
_Please describe the details of PullRequest._

## Types of changes

_What types of changes does your PullRequest introduce? Put an `x` in all the boxes that apply:_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Go over all the following points, and put an `x` in all the boxes that apply._

- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

**Comment by bergwolf** [comment]
Why is this a breaking change :)

**Comment by jiangliu** [comment]
> Why is this a breaking change :)

This PR introduces new crates, so existing code may fail to compile, so marked as "breaking changes".
It doesn't mean incompatibility,

File: builder/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+nydus-rafs = { version = "0.3", path = "../rafs" }
+nydus-storage = { version = "0.6", path = "../storage", features = ["backend-localfs"] }
+nydus-utils = { version = "0.4", path = "../utils" }
+
+[dev-dependencies]
+#vmm-sys-util = "0.10"
+#assert_matches = "1.5.0"
+
+[features]
+#fusedev = ["fuse-backend-rs/fusedev"]
```
Comment (imeoer) [inline builder/Cargo.toml:36]: Is it necessary to keep it?

Comment (jiangliu) [reply builder/Cargo.toml:36]: They should be removed:)

File: tests/texture/stargz/estargz_sample.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
+{
```
Comment (imeoer) [inline tests/texture/stargz/estargz_sample.json:1]: Is this file specially added?

Comment (jiangliu) [reply tests/texture/stargz/estargz_sample.json:1]: The builder has a ut case, which uses this file, though it's ignored currently.
```
   #[ignore]
    #[test]
    fn test_build_stargz_toc() {
        let tmp_dir = vmm_sys_util::tempdir::TempDir::new().unwrap();
        let tmp_dir = tmp_dir.as_path().to_path_buf();
        let root_dir = &std::env::var("CARGO_MANIFEST_DIR").expect("$CARGO_MANIFEST_DIR");
        let source_path =
            PathBuf::from(root_dir).join("../tests/texture/stargz/estargz_sample.json");
```

**Review by imeoer — APPROVED** [review]

**Pull request merged**